metrics-core/src/main/java/com/codahale/metrics/WeightedSnapshot.java

```diff
@@ -38,6 +38,12 @@ public WeightedSample(long value, double weight) {
      * @param values    an unordered set of values in the reservoir
      */
     public WeightedSnapshot(Collection<WeightedSample> values) {
+        if (values.size()==1 && Double.compare(values.iterator().next().weight,0D)==0) {
```

Comment:
I think it's not very obvious why we perform this check in the constructor of the snapshot. I understand that the idea to avoid the situation when we have only one element in the sample with a 0 weight (because it provides bogus data), but I believe it should be performed in `ExponentiallyDecayingReservoir`, where the logic of rescaling resides. What do you think, if we add a guard condition:
```java
if (scalingFactor == 0) {
    values.clear();
}
``` 
in the `ExponentiallyDecayingReservoir#rescale` method? I think it could be a more clear solution.